### PR TITLE
Add missing decorators to editorState when props update

### LINKED ISSUE #262: Editor Plugin doesn't react to a new editorState
when changing the editorState such as re-setting with `EditorState.createEmpty()`, a user would expect the plugin editor to react to the change in state as DraftJS's Editor does. But the plugin editor only binds the decorators in `componentWillMount` so the editor loses the plugins

### LINKED ISSUE #359: Inconsistency with decorators and EditorState
Using version `1.1.0`.

lets say we have this function:

```js
function createState(text) {
  return EditorState.createWithContent(
    ContentState.createFromText(text)
  );
}
```

each time we create a new `EditorState`, we need to manually set the decorators. Yet, the first state we send gets the decorators from the plugins.

### PR DESCRIPTION
**Pull request by juliankrispel**
## Checklist

- [x] Fix any eslint errors that occur
- [x] Update change-log for every plugin you touch
- [x] Add/Update tests if you add/change new functionality
- [x] Add/Update docs if you add/change functionality
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Tested in storybook

## Use-case/Problem
We add decorators to the editorState upon initalisation of the Editor but not when props update. This fixes this!

fixes #359 
fixes #262

## Implementation

- [x] reuse resolveDecorators function in willReceiveProps
- [x] add missing decorators to editorState object if editorState decorators are missing
- [x] refactor decorator resolving into separate function

### CODE REVIEW

**Comment by juliankrispel** [comment]
@mxstbr would be nice to have this reviewed by you since it specifically fixes a problem you're encountering

File: draft-js-plugins-editor/src/Editor/index.js [hunk 1]
```diff
@@ -1,22 +1,21 @@
...
 import proxies from './proxies';
+import moveSelectionToEnd from './moveSelectionToEnd';
+import resolveDecorators from './resolveDecorators';
 import * as defaultKeyBindingPlugin from './defaultKeyBindingPlugin';
 
 /**
  * The main editor component
  */
-class PluginEditor extends Component {
+class PluginEditor extends PureComponent {
```
Comment (mxstbr) [inline draft-js-plugins-editor/src/Editor/index.js]: Please don't make it a PureComponent: They can't have `ref`s, so you can't do `this.editor.focus()` and stuff like that.

Testing the patch now!

**Comment by mxstbr** [comment]
Testing this patch locally throws this error for me: https://reactjs.org/warnings/refs-must-have-owner.html

> addComponentAsRefTo(…): Only a ReactOwner can have refs. You might be adding a ref to a component that was not created inside a component’s render method, or you have multiple copies of React loaded.

**Comment by juliankrispel** [comment]
@mxstbr can you give it another test? It seems to work fine for me now :+1:

**Comment by mxstbr** [comment]
Still getting that error with the latest changes :confused: All we're doing is setting a `ref` on an editor.

**Comment by mxstbr** [comment]
Hmm, I think I might know why this is happening, we've actually got two copies of ReactDOM loaded on the page because for some reason linking locally duplicates the dependency...

**Comment by mxstbr** [comment]
Got this running locally by publishing a fork with these changes! Unfortunately it doesn't fix our issue, decorators are still not showing up :cry:

**Comment by juliankrispel** [comment]
@mxstbr WAT!!!!

I want to fix this use-case damnit! It's a really common one! Is there a chance I can see the implementation?

**Comment by juliankrispel** [comment]
@mxstbr ping me on slack or spectrum plz 👍

**Comment by mxstbr** [comment]
Update: This fixes the original issue perfectly, my comments were just me being dumb. This is awesome and so so helpful, ship it!

**Comment by juliankrispel** [comment]
@mxstbr https://www.npmjs.com/package/draft-js-plugins-editor 2.0.2

**Comment by mxstbr** [comment]
👌

**Pull request merged**